demos/mqtt/mqtt_demo_serializer/mqtt_demo_serializer.c

```diff
@@ -851,10 +890,20 @@ int main( int argc,
     NetworkContext_t networkContext = { 0 };
     RetryUtilsStatus_t retryUtilsStatus = RetryUtilsSuccess;
     RetryUtilsParams_t retryParams;
+    struct timespec tp;
 
     ( void ) argc;
     ( void ) argv;
 
+    /* Get current time to seed pseudo random number generator.
+     * Note: The random number generator is used for adding randomness
+     * to the client ID strings to avoid client identifier collision
+     * when connecting to the MQTT broker. */
+    ( void ) clock_gettime( CLOCK_REALTIME, &tp );
+
+    /* Seed pseudo random number generator with nanoseconds. */
+    srand( tp.tv_nsec );
```

Comment:
Is `srand(time(NULL));` not sufficient?

Reply:
Sure.

Reply:
On my embedded device tv_nsec is always 0 because my tick rate is one tick per ms. Would that mean every device always gets the same ID?